Jenkinsfile

```diff
@@ -1,13 +1,19 @@
 pipeline {
-    agent { label 'quick' }
+
+    agent {
+        dockerfile {
+            filename 'CI/Dockerfile.build'
+            label 'quick'
+        }
+    }
 
     options {
         timestamps ()
         timeout(time: 2, unit: 'HOURS')
     }
 
     environment {
-        GOROOT = '/usr/lib/go-1.21/'
+        GOROOT = '/usr/lib/go-1.22/'
```

Comment:
This line may not be needed. We installed a specific Go version in a custom directory on our vm, so we have to specify `GOROOT`.

Reply:
The point of this PR is to run the pipeline inside a container defined by the developers.  See line 4.
No tool in the host machine is to be used other than docker itself. This means the developers can install or remove versions of the toolchain without influencing other builds.

Reply:
Sorry bad explanation. I wanted to point out that `/usr/lib/go-1.21/` is a custom directory specific to the host machine. The directory may or may not exist in a docker container. Setting `GOROOT` to `usr/lib/go-1.22/`  is most likely incorrect.

Reply:
I am a little concerned about this line, but because of a different reason: 

I am confident that /usr/lib/go-1.21/ exists in the environment because docker is installed in the container image in jenkinsfile line 10.  The container will run in a dedicated filesystem, where the only visible folder from the host is the source code checkout. No other system or home folder from the host will be visible. Only the tools described in the jenkinsfile will be installed (in the default system paths). 

On the other hand: I copied this line from the current Jenkinsfile, which uses the current VM system, and this worries me, because it has been working fine so far. We should fix that, what do you propose?

Reply:
I tested creating the image on my local machine. It was my misunderstanding. Yes, `usr/lib/go-1.22` exists. I also noticed when running `go env` that `GOROOT` was automatically set to `usr/lib/go-1.22` on my container. The line may not be necessary. Can you try quickly by removing the line and see if the test still run successfully?